tests/results.cpp

```diff
@@ -932,12 +932,12 @@ TEST_CASE("notifications: sync") {
         // Start the server and wait for the Realm to be uploaded so that sync
         // makes some writes to the Realm and bumps the version
         server.start();
-        SyncManager::shared().get_session(config.path, *config.sync_config)->wait_for_upload_completion();
-
-        // Make sure that the notifications still get delivered rather than
-        // waiting forever due to that we don't get a commit notification from
-        // the commits sync makes to store the upload progress
-        r->refresh();
+        SyncManager::shared().get_session(config.path, *config.sync_config)->wait_for_upload_completion([r=std::move(r)](auto) {
+            // Make sure that the notifications still get delivered rather than
+            // waiting forever due to that we don't get a commit notification from
+            // the commits sync makes to store the upload progress
+            r->refresh();
+        });
```

Comment:
Since this no longer blocks I think you'd need to spin the runloop until the callback is called?